File: csharp/main/DecimalDef.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class DecimalDef : IConstantScope, ITypeDef<TypeDecimal>
	{
		private DecimalDef()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeDecimal.Instance, new LiteralDecimal(Decimal.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeDecimal.Instance, new LiteralDecimal(Decimal.MaxValue)));
			DefinedConstants.Add(new ConstantDef(this, "NaN", TypeDecimal.Instance, new LiteralDecimal("Decimal.NaN")));
			DefinedConstants.Add(new ConstantDef(this, "POSITIVE_INFINITY", TypeDecimal.Instance, new LiteralDecimal("Decimal.POSITIVE_INFINITY")));
			DefinedConstants.Add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeDecimal.Instance, new LiteralDecimal("Decimal.NEGATIVE_INFINITY")));
			DefinedConstants.Add(new ConstantDef(this, "NULL", TypeDecimal.Instance, new LiteralDecimal("Decimal.NULL")));
		}

		public static DecimalDef Instance = new DecimalDef();

		public String Name => "Decimal";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeDecimal Type => TypeDecimal.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}

File: java/src/main/java/deltix/luminary/DecimalDef.java
package deltix.luminary;

import deltix.dfp.Decimal64Utils;

import java.util.ArrayList;
import java.util.List;

public class DecimalDef implements TypeDef<TypeDecimal>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private DecimalDef() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.NEGATIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.POSITIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "NaN", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.NaN)));
        definedConstants.add(new ConstantDef(this, "POSITIVE_INFINITY", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.POSITIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "NEGATIVE_INFINITY", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.NEGATIVE_INFINITY)));
        definedConstants.add(new ConstantDef(this, "NULL", TypeDecimal.INSTANCE, new LiteralDecimal(Decimal64Utils.NULL)));
    }

    public static final DecimalDef INSTANCE = new DecimalDef();

    @Override
    public String getName() {
        return "Decimal";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeDecimal getType() {
        return TypeDecimal.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}
